### PR TITLE
GM Security Policy Support

### PR DESCRIPTION
**Pull request by enhaocui**

### CODE REVIEW

File: nsxt/resource_nsxt_policy_security_policy_test.go [hunk 1]
```diff
@@ -240,7 +350,16 @@ func testAccNsxtPolicySecurityPolicyExists(resourceName string, domainName strin
 			return fmt.Errorf("Policy SecurityPolicy resource ID not set in resources")
 		}
 
-		_, err := nsxClient.Get(domainName, resourceID)
+		var err error
+		if isPolicyGlobalManager(testAccProvider.Meta()) {
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_security_policy_test.go:354]: can we use resourceNsxtPolicySecurityPolicyExistsInDomain here?

Comment (enhaocui) [reply nsxt/resource_nsxt_policy_security_policy_test.go:354]: Thanks for the catch. Yes we can, but that way we will lose track of the error message here. Instead the error will be logged in logAPIError in resourceNsxtPolicySecurityPolicyExistsInDomain. Do you think it's OK here?

Comment (annakhm) [reply nsxt/resource_nsxt_policy_security_policy_test.go:354]: Yes, I think its OK since we don't expect errors other than 404 at this stage

Comment (enhaocui) [reply nsxt/resource_nsxt_policy_security_policy_test.go:354]: Sure. Let me fix this here. Thanks!

File: nsxt/resource_nsxt_policy_security_policy_test.go [hunk 2]
```diff
@@ -240,35 +350,64 @@ func testAccNsxtPolicySecurityPolicyExists(resourceName string, domainName strin
...
 	for _, rs := range state.RootModule().Resources {
 
 		if rs.Type != "nsxt_policy_security_policy" {
 			continue
 		}
 
 		resourceID := rs.Primary.Attributes["id"]
-		_, err := nsxClient.Get(domainName, resourceID)
+		if isPolicyGlobalManager {
+			_, err = nsxClient.(*gm_domains.DefaultSecurityPoliciesClient).Get(domainName, resourceID)
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_security_policy_test.go:379]: I think we can use resourceNsxtPolicySecurityPolicyExistsInDomain here as well?

Comment (enhaocui) [reply nsxt/resource_nsxt_policy_security_policy_test.go]: Thanks for the catch! fixed.

**Pull request merged**